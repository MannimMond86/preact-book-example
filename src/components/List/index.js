import { h, Component } from 'preact';
import Item from '../Item';

export default class List extends Component {
	render() {
		return (
			<ol>
				{this.props.items.map((item, idx) => <Item index={idx + 1} item={item} />)}
			</ol>
		);
	}
}
