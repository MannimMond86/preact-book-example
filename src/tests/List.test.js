import { h, Component } from 'preact';
import List from '../components/List';

describe('<List />', () => {
	it('should a list of articles', () => {
		const items = [
			{ title: 'post 1', time_ago: '1 minute ago' },
			{ title: 'post 2', time_ago: '2 minutes ago' },
			{ title: 'post 3', time_ago: '3 minutes ago' },
		];
		expect(<List items={items} />).toMatchSnapshot();
	});
});

