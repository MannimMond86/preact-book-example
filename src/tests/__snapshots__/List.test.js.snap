// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should a list of articles 1`] = `
e {
  "attributes": Object {
    "items": Array [
      Object {
        "time_ago": "1 minute ago",
        "title": "post 1",
      },
      Object {
        "time_ago": "2 minutes ago",
        "title": "post 2",
      },
      Object {
        "time_ago": "3 minutes ago",
        "title": "post 3",
      },
    ],
  },
  "children": Array [],
  "key": undefined,
  "nodeName": [Function],
}
`;
